_posts/2019-07-16-gentle-intro-to-docker.markdown

```diff
@@ -0,0 +1,65 @@
+
+How about something that's a little more fun to play with?  Run:
+
+``docker run -it ubuntu bash``
+
+Just like before, Docker will download an image - this time, the well known Linux variant Ubuntu - and then create a container to run it in.  Note the extra argument that we added at the end, though - a command that will be executed within the downloaded image.  It's the same basic syntax (and idea) that you would use to run a command remotely with ssh.  In this case, it'll give you a nice, interactive shell that lets you poke around in your container.  The ``-it`` bit is a pair of command line options that allocate a terminal and tell Docker that you want to actually interact with the container.  Now try:
+
+``cd /``
+
+And notice that you're *not* in your own root directory, but the container's.  Try as you might, you can't get out to your host file system - the container is isolated from the rest of your OS.  The only way to get out is to exit the shell. Try creating a few files, then exiting the container and running it again. You'll see that they're gone - any changes you make to the container are lost once you exit.  This is very useful for creating temporary, disposable development and testing environments!
+
+To really drive home the difference between an image and a container, try running the Ubuntu command above in a few terminals at once.  Then try making some modifications, like creating or deleting some files.  If you look in the other running Ubuntu containers, you'll see that your changes are not reflected in them.  They're isolated, remember?  You can see a list of all the containers you're running by executing:
+
+``docker ps``
+
+And what about that hello-world image that we started out with?  It would be nice if we could open that up and take a look at it's guts, don't you think?  There's one complication, though: it doesn't include a shell like Bash or sh that can run to get an interactive session going.  Wouldn't it be nice if we could mount the image as a file system and use another image with the tools we need to poke around?  We can!
+
+To do that, we're going to download the [Busybox](https://busybox.net) image, an image that rolls a bunch of standard Linux utilities into a single executable, described as the "swiss army knife" of Docker images.  Once we mount the container on ``/tmp/dockerimage`` we'll open a shell that will let us inspect the mounted image.  Start by running:
+
+``docker run -it -v=hello-world:/tmp/dockerimage busybox sh``
+
+If everything goes smoothly, you'll end up with a shell prompt running within a hello-world container - poke around and see what you can find.  As an added bonus, if you'd like to see a dump of a whole bunch of information about the image, try:
+
+``docker image inspect hello-world``
+
+At this point you're starting to build up a nice little collection of images.  You can see them with:
+
+``docker image ls``
```

Comment:
you can also use `docker images` but i think that might depend on the docker version.